src/wp-admin/user-edit.php

```diff
@@ -546,77 +561,108 @@
 						</td>
 					</tr>
 
-					<tr class="user-url-wrap">
-						<th><label for="url"><?php _e( 'Website' ); ?></label></th>
-						<td><input type="url" name="url" id="url" value="<?php echo esc_attr( $profile_user->user_url ); ?>" class="regular-text code" /></td>
-					</tr>
+					<?php
+					/*
+					 * Filters the user Contact Methods section.
+					 *
+					 * @param bool    Whether to show the contact methods fields. Default true.
+					 *
+					 * @since CP-2.0.0
+					 */
+					if ( apply_filters( 'user_profile_contact_methods', true ) ) :
+					?>
+
+						<tr class="user-url-wrap">
+							<th><label for="url"><?php _e( 'Website' ); ?></label></th>
+							<td><input type="url" name="url" id="url" value="<?php echo esc_attr( $profile_user->user_url ); ?>" class="regular-text code" /></td>
+						</tr>
+
+						<?php foreach ( wp_get_user_contact_methods( $profile_user ) as $name => $desc ) : ?>
+						<tr class="user-<?php echo $name; ?>-wrap">
+							<th>
+								<label for="<?php echo $name; ?>">
+								<?php
+								/**
+								 * Filters a user contactmethod label.
+								 *
+								 * The dynamic portion of the hook name, `$name`, refers to
+								 * each of the keys in the contact methods array.
+								 *
+								 * @since 2.9.0
+								 *
+								 * @param string $desc The translatable label for the contact method.
+								 */
+								echo apply_filters( "user_{$name}_label", $desc );
+								?>
+								</label>
+							</th>
+							<td>
+								<input type="text" name="<?php echo $name; ?>" id="<?php echo $name; ?>" value="<?php echo esc_attr( $profile_user->$name ); ?>" class="regular-text" />
+							</td>
+						</tr>
+
+						<?php
+						endforeach;
+					endif; // end 'user_profile_contact_methods' filter
+					?>
 
-					<?php foreach ( wp_get_user_contact_methods( $profile_user ) as $name => $desc ) : ?>
-					<tr class="user-<?php echo $name; ?>-wrap">
-						<th>
-							<label for="<?php echo $name; ?>">
-							<?php
-							/**
-							 * Filters a user contactmethod label.
-							 *
-							 * The dynamic portion of the hook name, `$name`, refers to
-							 * each of the keys in the contact methods array.
-							 *
-							 * @since 2.9.0
-							 *
-							 * @param string $desc The translatable label for the contact method.
-							 */
-							echo apply_filters( "user_{$name}_label", $desc );
-							?>
-							</label>
-						</th>
-						<td>
-							<input type="text" name="<?php echo $name; ?>" id="<?php echo $name; ?>" value="<?php echo esc_attr( $profile_user->$name ); ?>" class="regular-text" />
-						</td>
-					</tr>
-					<?php endforeach; ?>
 				</table>
 
 				<h2><?php IS_PROFILE_PAGE ? _e( 'About Yourself' ) : _e( 'About the user' ); ?></h2>
```

Comment:
Maybe this line can be changed to something like 
```php
				<?php
				if ( apply_filters( 'user_profile_about_yourself', true ) || get_option( 'show_avatars' ) ) {
					echo '<h2>';
					IS_PROFILE_PAGE ? _e( 'About Yourself' ) : _e( 'About the user' );
					echo '</h2>';
				}
				?>
```
This way if the avatar is off and the about section is hidden the header is not shown without content.

Reply:
Great idea! Now done!